src/storage/heap_file.c

```diff
@@ -14530,23 +14531,24 @@ heap_dump_heap_file (THREAD_ENTRY * thread_p, FILE * fp, bool dump_records, cons
   status = xlocator_find_class_oid (thread_p, class_name, &class_oid, S_LOCK);
   if (status != LC_CLASSNAME_EXIST)
     {
-      return;
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_LC_UNKNOWN_CLASSNAME, 1, class_name);
+      return ER_LC_UNKNOWN_CLASSNAME;
     }
 
+  fprintf (fp, "\n*** DUMP HEAP OF %s ***\n", class_name);
+
   error_code = heap_hfid_cache_get (thread_p, &class_oid, &hfid, NULL, NULL);
   if (error_code != NO_ERROR)
     {
-      assert (false);
-      return;
+      return error_code;
```

Comment:
```suggestion
      assert (false);
      return error_code;
```
Since it should never happen, it is good to leave the assert(false) as it is. Let's use this to check for issues in debug build mode.